app/decorators/song_decorator.rb

```diff
@@ -0,0 +1,11 @@
+class SongDecorator < Draper::Decorator
+  delegate_all
+
+  def duration_in_minutes_and_seconds
+    "#{duration / 60}:#{duration % 60}"
+  end
+
+  def full_title
+    "#{song.artist.name} - #{song.title} - #{song.album}"
```

Comment:
Instead of referencing `song`, you'd actually want to reference `object` as the Draper gem has the song as its `object`.

See: https://github.com/drapergem/draper#accessing-the-model